controllers/secret_controller.go

```diff
@@ -162,36 +163,58 @@ func (r *SecretReconciler) Reconcile(ctx context.Context, request ctrl.Request)
 		// valid userData secret already exists
 		return reconcile.Result{}, nil
 	} else {
-		// userdata secret data does not match what is expected
-		// Mark nodes configured with the previous private key for deletion
+		// userdata secret data does not match what is expected, implies private key has changed
+		// Modify annotations on nodes configured with the previous private key
 		nodes := &core.NodeList{}
 		err = r.client.List(ctx, nodes, client.MatchingLabels{core.LabelOSStable: "windows"})
 		if err != nil {
 			return reconcile.Result{}, errors.Wrapf(err, "error getting node list")
 		}
+
 		expectedPubKeyAnno := nodeconfig.CreatePubKeyHashAnnotation(keySigner.PublicKey())
-		patchData, err := annotations.GenerateAddPatch(map[string]string{nodeconfig.PubKeyHashAnnotation: ""})
+		privateKeyBytes, err := secrets.GetPrivateKey(kubeTypes.NamespacedName{Namespace: r.watchNamespace,
+			Name: secrets.PrivateKeySecret}, r.client)
 		if err != nil {
-			return reconcile.Result{}, errors.Wrapf(err, "error creating public key annotation add request")
+			return reconcile.Result{}, err
 		}
 		for _, node := range nodes.Items {
-			// Only clear the annotation for Nodes associated with Machines, as the clearing of the annotation is used
-			// solely to kick off the deletion and recreation of Machines.
+			annotationsToApply := make(map[string]string)
 			if _, present := node.Annotations[BYOHAnnotation]; present {
-				continue
-			}
+				// For BYOH nodes, update the username annotation and public key hash annotation using new private key
+				instanceUsername, err := r.usernameFromNode(ctx, &node, nodes)
+				if err != nil {
+					return reconcile.Result{}, err
+				}
+				usernameCipher, err := secrets.Encrypt(instanceUsername, privateKeyBytes)
+				if err != nil {
+					return reconcile.Result{}, err
+				}
+				// Obfuscate newline characters to make encrypted data compatible as JSON Patch request body
+				patchCombatibleCipher := strings.Replace(usernameCipher, "\n", "<newline>", -1)
```

Comment:
You have repeated code here and converting these lines into a helper method will fix that like I called out before.